spec.json

```diff
@@ -1,18 +1,202 @@
 {
   "nio/ESFind": {
-    "version": "0.0.1",
-    "description": "",
-    "inputs": {},
-    "outputs": {},
-    "properties": {},
-    "commands": {}
+    "version": "0.1.0",
+    "description": "Finds elements from given search parameters.",
+    "properties": {
+      "auth": {
+        "title": "Authentication",
+        "type": "ObjectType",
+        "description": "Username and password credentials to connect to the Elastic Search database.",
+        "default": {
+          "use_https": false,
+          "username": "",
+          "password": ""
+        }
+      },
+      "condition": {
+        "title": "Condition",
+        "type": "Type",
+        "description": "Condition to filter data on.",
+        "default": "{'match_all': {}}"
+      },
+      "doc_type": {
+        "title": "Type",
+        "type": "Type",
+        "description": "The type of the document to query.",
+        "default": "{{($__class__.__name__)}}"
+      },
+      "elasticsearch_client_kwargs": {
+        "title": "Client Argurments",
+        "type": "Type",
+        "description": "kwargs to be passed to elasticsearch client. (e.g.: {'maxsize': 15})",
+        "default": null
+      },
+      "enrich": {
+        "title": "Signal Enrichment",
+        "type": "ObjectType",
+        "description": "*enrich_field:* The attribute on the signal to store the results from this block. If this is empty, the results will be merged onto the incoming signal. This is the default operation. Having this field allows a block to 'save' the results of an operation to a single field on an incoming signal and notify the enriched signal.  *results field:* The attribute on the signal to store the results from this block. If this is empty, the results will be merged onto the incoming signal. This is the default operation. Having this field allows a block to save the results of an operation to a single field on an incoming signal and notify the enriched signal.",
```

Comment:
Noticed a similar comment somewhere else... this is what we're doing for the enrich mixin?